pkg/slices/slices.go

```diff
@@ -0,0 +1,20 @@
+package slices
+
+// Chunk chunks a slice into batches of chunkSize.
+// example: {1,2,3,4,5}, chunkSize = 2 -> {1,2}, {3,4}, {5}
+func Chunk[T any](input []T, chunkSize int) [][]T {
```

Comment:
taken from sauron. 

There is a [Chunk](https://pkg.go.dev/slices#Chunk) function in the `slices` pkg for go1.23, but this would require is to bump the go version in our go.mod, which would affect anyone using our code